File: 1258-59-2375-construct-smallest-number-from-di-string/1258-59-2375-construct-smallest-number-from-di-string.cpp
class Solution {
public:
    string smallestNumber(string pattern) {
        int n = pattern.size();
        string stk,ans;
        for(int i = 0 ; i <= n;i++){
            stk.push_back((i+1)+'0');
            if(i == n || pattern[i] == 'I')
                while(stk.size())ans.push_back(stk.back()),stk.pop_back();
        }
        return ans;
    }
};

File: 1258-59-2375-construct-smallest-number-from-di-string/1258-59-2375-construct-smallest-number-from-di-string.js
var smallestNumber = function(pattern) {
    let n = pattern.length, stk = [], ans = [];
    for (let i = 0; i <= n; i++) {
        stk.push((i + 1).toString());
        if (i === n || pattern[i] === 'I') {
            while (stk.length) ans.push(stk.pop());
        }
    }
    return ans.join('');
};
